sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -87,8 +87,7 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
       case "overwrite" => mode(SaveMode.Overwrite)
       case "append" => mode(SaveMode.Append)
       case "ignore" => mode(SaveMode.Ignore)
-      case "error" | "errorifexists" => mode(SaveMode.ErrorIfExists)
-      case "default" => this
+      case "error" | "errorifexists" | "default" => mode(SaveMode.ErrorIfExists)
       case _ => throw new IllegalArgumentException(s"Unknown save mode: $saveMode. " +
         "Accepted save modes are 'overwrite', 'append', 'ignore', 'error', 'errorifexists'.")
```

Comment:
Not related to the PR but the list could include `"default"` as well.

Reply:
Got it~